### ISSUE TITLE
Solr range facet data issues

### ISSUE THREAD

**Issue by thatbudakguy** [Princeton-CDH/ppa-django#131]
Other things appear to be broken with the histogram, however, as I noted on Slack. 
...
(2) If the year span is small enough(?) when doing a keyword search the histogram only shows bars for the start and end date, and it's blank in between. That doesn't seem right? Here are two screenshots of the search "iamb" with different year spans.
<img width="735" alt="screen shot 2018-10-25 at 3 08 10 pm" src="https://user-images.githubusercontent.com/22945600/47525465-52612a80-d86b-11e8-8f3a-a76f0d754395.png">
<img width="720" alt="screen shot 2018-10-25 at 3 11 57 pm" src="https://user-images.githubusercontent.com/22945600/47525469-542aee00-d86b-11e8-9fe0-35487dbfbe6b.png">

_Originally posted by @mnaydan in https://github.com/Princeton-CDH/ppa-django/issues/95#issuecomment-433178280_

**Comment by thatbudakguy** [comment]
update: histogram isn't displaying anything at all in production.

**Comment by rlskoeser** [comment]
Tested the facet options on solr directly, and we're getting the zeros because of the `facet.range.include` option we're setting to 'outer'; it only happens when `gap` is set to 1.  

Here's the commit where we changed it: https://github.com/Princeton-CDH/ppa-django/commit/4b1e0a0d6d1a9c6762a185cd10a767d633e0fc3e 
@thatbudakguy do you remember this? What do we need to do to go back to using `end + 1` and drop the include option?

**Comment by mnaydan** [comment]
@thatbudakguy so this generally works now, but if you enter certain invalid ranges and then enter a valid range the histogram for the valid range remains blank even after resubmitting the form. Two ways to produce this: If you use the up/down arrows on the max range, it still defaults to 1559, so the min value is bigger than the max value. This would be fixed by fixing #138. See screenshots:
<img width="1008" alt="screen shot 2018-11-29 at 2 45 59 pm" src="https://user-images.githubusercontent.com/22945600/49247634-c4d2a680-f3e5-11e8-800c-2897921917b1.png">
<img width="1008" alt="screen shot 2018-11-29 at 2 46 12 pm" src="https://user-images.githubusercontent.com/22945600/49247639-c69c6a00-f3e5-11e8-961a-1e92faf03d36.png">

But you still have a problem if you manually input a bigger value in the min field and a smaller value in the max field, and then try to make the values valid: the form will submit and the histogram labels change but the histogram stays blank, even though the new values are valid. If you refresh the page with an invalid range and then try to make them valid it gets even wackier (see screenshot).
<img width="1011" alt="screen shot 2018-11-29 at 2 44 16 pm" src="https://user-images.githubusercontent.com/22945600/49247762-12e7aa00-f3e6-11e8-9d73-b172d61f512b.png">

**Comment by thatbudakguy** [comment]
@mnaydan I think you identified a wholly new (well, newly discovered) bug, which I broke off into #186.

Based on your comments I think this particular issue of the histogram only showing start/end date bars is fixed, however. Are you OK to close it?

**Comment by mnaydan** [comment]
@thatbudakguy yes, the way you phrased it in #186 is perfect... I'll close out.
